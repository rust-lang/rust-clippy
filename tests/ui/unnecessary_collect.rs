#![warn(clippy::unnecessary_collect)]
//@no-rustfix

fn bad() -> Vec<u32> {
    (0..5).collect()
    //~^ unnecessary_collect
}
unsafe fn bad2() -> Vec<(u8, u8)> {
    (0..8).flat_map(|x| (0..8).map(move |y| (x, y))).collect()
    //~^ unnecessary_collect
}
fn okay() -> String {
    ('a'..='z').collect()
}
fn hmm() -> std::collections::HashSet<u32> {
    (0..5).chain(3..12).collect()
}
fn good() -> impl Iterator<Item = u32> {
    (5..10).rev()
}
fn main() {}
