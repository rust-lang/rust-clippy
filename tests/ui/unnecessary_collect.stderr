error: this collection may not be necessary. return the iterator itself
  --> tests/ui/unnecessary_collect.rs:5:12
   |
LL | fn bad() -> Vec<u32> {
   |             -------- help: consider: `impl Iterator<Item = u32>`
LL |     (0..5).collect()
   |            ^^^^^^^^^ remove this collect
   |
   = note: `-D clippy::unnecessary-collect` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::unnecessary_collect)]`

error: this collection may not be necessary. return the iterator itself
  --> tests/ui/unnecessary_collect.rs:9:54
   |
LL | unsafe fn bad2() -> Vec<(u8, u8)> {
   |                     ------------- help: consider: `impl Iterator<Item = (u8, u8)>`
LL |     (0..8).flat_map(|x| (0..8).map(move |y| (x, y))).collect()
   |                                                      ^^^^^^^^^ remove this collect

error: aborting due to 2 previous errors

